scripts/perf/deploy_instance/run_on_instance.sh

```diff
@@ -0,0 +1,51 @@
+#! /bin/bash
+
+set -x
+
+### /!\ This script assumes docker and docker compose are already installed on the host
+
+# Utility function to extract values from instance metadata
+extract_metadata() {
+  curl "http://metadata.google.internal/computeMetadata/v1/instance/attributes/$1" -H "Metadata-Flavor: Google"
```

Comment:
ToL this looks familiar.   Seeing a possibility for a library around vault/GCE metadata/env vars